xea-web/src/components/login/LoginForm.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+import { connect } from 'react-redux'
+import { Field, reduxForm } from 'redux-form'
+import { FormGroup, FormControl, ControlLabel, HelpBlock } from 'react-bootstrap'
+
+import ErrorMessage from '../common/ErrorMessage'
+
+class LoginForm extends Component {
+  constructor (props) {
+    super(props)
+
+    this.getValidationState = this.getValidationState.bind(this)
+    this.renderField = this.renderField.bind(this)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    // this.username.focus()
+    const username = this.username
+    /* eslint-disable react/no-find-dom-node */
+    ReactDOM.findDOMNode(username).focus()
+    /* eslint-enable react/no-find-dom-node */
+  }
+
+  getValidationState ({ pristine, active, error }) {
+    if (pristine) {
+      return 'success'
+    }
+    if (error) {
+      if (active) {
+        return 'warning'
+      }
+      return 'error'
+    }
+    return 'success'
+  }
+
+  usernameNormalicer (value, previousValue) {
+    // Prevent user form use whitespaces
+    if (/\s+/.test(value)) {
+      return previousValue
+    }
+    return value
+  }
+
+  renderField ({ name, placeholder, input, label, type,
```

Comment:
O "normalizer"?
Igual poderían facerse "normalizers" para distintas cousas que se puidesen combinar é metelos todos nunha biblioteca. Se tal mellor esperar a ir atopando a necesidade de normalizar máis formularios?